cgo_static.go

```diff
@@ -4,9 +4,10 @@ package duckdb_go_extensions
 
 ////-lsubstrait_extension
 /*
-#cgo LDFLAGS: -licu_extension -ltpch_extension -ltpcds_extension -lsubstrait_extension
-#cgo darwin,amd64 LDFLAGS: -lc++ -L${SRCDIR}/deps/darwin_amd64
-#cgo darwin,arm64 LDFLAGS: -lc++ -L${SRCDIR}/deps/darwin_arm64
+#cgo LDFLAGS: -licu_extension -ltpch_extension -ltpcds_extension -lsubstrait_extension -liceberg_extension
```

Comment:
what does the output of `ldd` say for the resulting duckdb/go lib? I can't tell what are shared and what are static libraries.

I am most worried that our deployment environment is expecting boost or avro shared objects